libraries/opensk/src/api/key_store.rs

```diff
@@ -12,12 +12,60 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::api::crypto::aes256::Aes256;
 use crate::api::crypto::ecdsa::SecretKey as _;
+use crate::api::crypto::hmac256::Hmac256;
+use crate::api::crypto::HASH_SIZE;
+use crate::ctap::crypto_wrapper::{aes256_cbc_decrypt, aes256_cbc_encrypt, PrivateKey};
+use crate::ctap::data_formats::CredentialProtectionPolicy;
 use crate::ctap::secret::Secret;
-use crate::env::{EcdsaSk, Env};
+use crate::ctap::{cbor_read, cbor_write};
+use crate::env::{AesKey, EcdsaSk, Env, Hmac};
 use alloc::vec;
+use alloc::vec::Vec;
+use core::convert::{TryFrom, TryInto};
 use persistent_store::StoreError;
 use rand_core::RngCore;
+use sk_cbor as cbor;
+use sk_cbor::{cbor_map_options, destructure_cbor_map};
+
+pub const LEGACY_CREDENTIAL_ID_SIZE: usize = 112;
+// CBOR credential IDs consist of
+// - 1   byte : version number
+// - 16  bytes: initialization vector for AES-256,
+// - 192 bytes: encrypted block of the key handle cbor,
+// - 32  bytes: HMAC-SHA256 over everything else.
+pub const CBOR_CREDENTIAL_ID_SIZE: usize = 241;
+pub const MIN_CREDENTIAL_ID_SIZE: usize = LEGACY_CREDENTIAL_ID_SIZE;
+pub const MAX_CREDENTIAL_ID_SIZE: usize = CBOR_CREDENTIAL_ID_SIZE;
+
+pub const CBOR_CREDENTIAL_ID_VERSION: u8 = 0x01;
+
+pub const MAX_PADDING_LENGTH: u8 = 0xBF;
```

Comment:
Why do all those constants need to be public?

Reply:
Some don't, fixed. The remaining are exposed in GetInfo or used in CTAP1.

Reply:
But those are not part of the API. We should use `pub(crate)` then.

Reply:
Being strict with `pub` and `pub(crate)` is probably a bigger chance also, that clearly divides what is supposed to be in the API and what isn't. Probably as much effort as hiding `src/ctap` from the outside and re-exporting in `src/lib` only what we need, and doing a proper API design.

Reply:
I'm not saying to `pub(crate)` everything that should be `pub(crate)`, I'm just saying we should not make those constant public in this PR while they were not public before. None of those constants should be in the API as far as I can tell.

Reply:
If you implement your own credential wrapping, but want to add padding to match the standard size, then `CBOR_CREDENTIAL_ID_SIZE` would be useful.

Reply:
Yes good point. Let's keep that one and mention it explicitly in the documentation of the wrap function.